From b3975c0f4ca66ffa2cd55527b397a76431c511f1 Mon Sep 17 00:00:00 2001
From: Martin Storsjo <martin@martin.st>
Date: Mon, 23 Sep 2019 12:03:08 +0000
Subject: [UPSTREAM][LLDB] Add a missing specification of linking against
 dbghelp

The PECOFF object file plugin uses the dbghelp API, but doesn't
specify that it has to be linked in anywhere.

Current MSVC based builds have probably succeeded, as other parts
in LLDB have had a "#pragma comment(lib, "dbghelp.lib")", but there's
currently no such pragma in the PECOFF plugin.

The "#pragma comment(lib, ...)" approach doesn't work in MinGW mode
(unless the compiler is given the -fms-extensions option, and even
then, it's only supported by clang/lld, not by GCC/binutils), thus
add it to be linked via CMake. (The other parts of LLDB that use
dbghelp are within _MSC_VER ifdefs.)

Differential Revision: https://reviews.llvm.org/D67885

llvm-svn: 372587
(cherry picked from commit fb31852fbc4e3095527180f0aef655800f3f04d4)

Change-Id: Ie02eaf3159b883dde2c6c1674d3e7385480b0e4f
---
 lldb/source/Plugins/ObjectFile/PECOFF/CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lldb/source/Plugins/ObjectFile/PECOFF/CMakeLists.txt b/lldb/source/Plugins/ObjectFile/PECOFF/CMakeLists.txt
index 04321f27655..6981f7ecb71 100644
--- a/lldb/source/Plugins/ObjectFile/PECOFF/CMakeLists.txt
+++ b/lldb/source/Plugins/ObjectFile/PECOFF/CMakeLists.txt
@@ -1,3 +1,10 @@
+# Dbghelp is used on windows for writing minidump files.
+if(WIN32)
+  set(DBGHELP_LINK_FILES dbghelp)
+else()
+  set(DBGHELP_LINK_FILES "")
+endif()
+
 add_lldb_library(lldbPluginObjectFilePECOFF PLUGIN
   ObjectFilePECOFF.cpp
   WindowsMiniDump.cpp
@@ -7,6 +14,7 @@ add_lldb_library(lldbPluginObjectFilePECOFF PLUGIN
     lldbHost
     lldbSymbol
     lldbTarget
+    ${DBGHELP_LINK_FILES}
   LINK_COMPONENTS
     BinaryFormat
     Support
-- 
2.25.0.341.g760bfbb309-goog

